docs/js/sandbox.js

```diff
@@ -0,0 +1,77 @@
+import LZString from "../vendor/lz-string/lz-string.js";
+
+// The key used to store the sandbox in local/session storage
+export const codeStorageKey = `sandbox-code`;
+
+export const saveSandboxCodeToStorage = (code) => {
+  localStorage.setItem(codeStorageKey, code);
+}
+
+// The example grammar to use when there is no saved code in the URL or local storage
+export const exampleGrammar = `
+// Simple Arithmetics Grammar
+// ==========================
+//
+// Accepts expressions like "2 * (3 + 4)" and computes their value.
+
+Expression
+  = head:Term tail:(_ ("+" / "-") _ Term)* {
+      return tail.reduce(function(result, element) {
+        if (element[1] === "+") { return result + element[3]; }
+        if (element[1] === "-") { return result - element[3]; }
+      }, head);
+    }
+
+Term
+  = head:Factor tail:(_ ("*" / "/") _ Factor)* {
+      return tail.reduce(function(result, element) {
+        if (element[1] === "*") { return result * element[3]; }
+        if (element[1] === "/") { return result / element[3]; }
+      }, head);
+    }
+
+Factor
+  = "(" _ expr:Expression _ ")" { return expr; }
+  / Integer
+
+Integer "integer"
+  = _ [0-9]+ { return parseInt(text(), 10); }
+
+_ "whitespace"
+  = [ \\t\\n\\r]*
+`;
+
+/**
+ * @param {URL} url
+ * @returns {string | null}
+ */
+export function getSandboxInitialContents(url) {
+  if (url.hash.startsWith("#code/")) {
```

Comment:
Needs to get sample input as well.

Reply:
Makes sense, I changed this so it now captures the whole state as an object, so we can decide to add more things later if we want.